lib/helper/REST.js

```diff
@@ -220,5 +220,188 @@ class REST extends Helper {
+ * Checks if the response code is equal to what was provided
+ *
+ * ```js
+ * I.seeResponseCodeIs(200);
+ * ```
+ *
+ * @param {string|number} statusCode
+ */
+  seeResponseCodeIs(statusCode) {
+    expect(this.response.status).to.equal(statusCode, `Status code ${this.response.status} of response is not equal to ${statusCode}`);
+  }
+
+  /**
+   * Checks if the response code is not equal to what was provided
+   *
+   * ```js
+   * I.dontSeeResponseCodeIs(200);
+   * ```
+   *
+   * @param {string|number} statusCode
+   */
+  dontSeeResponseCodeIs(statusCode) {
+    expect(this.response.status).to.not.equal(statusCode, `Status code ${this.response.status} of response is equal to ${statusCode}`);
+  }
+
+  /**
+   * Checks if the response code is a sucessful response code by checking if its value is between 200-299
+   *
+   * ```js
+   * I.seeResponseWasSucessful();
+   * ```
+   */
+  seeResponseWasSucessful() {
+    expect(this.response.status).to.be.above(199);
+    expect(this.response.status).to.be.below(300);
+  }
+
+  /**
+   * Checks if the response code is a redirect response code by checking if its value is between 300-399
+   *
+   * ```js
+   * I.seeResponseWasRedirected();
+   * ```
+   */
+  seeResponseWasRedirected() {
+    expect(this.response.status).to.be.above(299);
+    expect(this.response.status).to.be.below(400);
+  }
+
+  /**
+   * Checks if the response code is an error response code by checking if its value is between 400-499
+   *
+   * ```js
+   * I.seeResponseWasClientError();
+   * ```
+   */
+  seeResponseWasClientError() {
+    expect(this.response.status).to.be.above(399);
+    expect(this.response.status).to.be.below(500);
+  }
+
+  /**
+   * Checks if the response code is a server error response code by checking if its value is between 500-599
+   *
+   * ```js
+   * I.seeResponseWasServerError();
+   * ```
+   */
+  seeResponseWasServerError() {
+    expect(this.response.status).to.be.above(499);
+    expect(this.response.status).to.be.below(600);
+  }
+
+  /**
+   * Checks if the response code is between a specified range
+   *
+   * ```js
+   * I.seeResponseCodeBetween(200, 299);
+   * ```
+   */
+  seeResponseCodeBetween(min, max) {
+    expect(this.response.status).to.be.above(min);
+    expect(this.response.status).to.be.below(max);
+  }
+
+  /**
+   * Checks if a header exists on the request and optionally checks the header's value
+   *
+   * ```js
+   * I.seeHeader('Content-Type');
+   * I.seeHeader('Content-Type', 'application/json');
+   * ```
+   *
+   * @param {string} key
+   * @param {string} [value=undefined]
+   */
+  seeHeader(key, value = undefined) {
+    expect(this.response.headers).to.have.property(key);
+    if (value) {
+      expect(this.response.headers[key]).to.equal(value);
+    }
+  }
+
+  /**
+   * Checks if a header does not exist on the request
+   *
+   * ```js
+   * I.dontSeeHeader('Content-Type');
+   * ```
+   *
+   * @param {string} key
+   */
+  dontSeeHeader(key) {
```

Comment:
can we add value like we did for `seeHeader` or isn't it giving no benefits?
`dontSeeHeader(key, value = undefined)`

Reply:
Ah, so like you are wanting to assert a particular header doesn't have a certain value? I can look into that

Reply:
yeah, like we provide more flexible assertion, user can either assert `key` or `key:value`

Reply:
I guess you can't use the 2 step process I have for see header where you

1. check if the header exists
2. check if the value of that header is what the user provided

it would instead be

if only key --> check that key doesn't exist as a header
if key+value --> check that header exists and then check that the value provided doesn't exist on the header

Make sense?

Reply:
yes, indeed! 😄